SJQ2T-21: Forgot password functionality

DraganMitrasinovicHtec opened the pull request:
Added the forgot password functionality, user can now request a password reset email.
After getting the email, he will be able to go to that page and enter a new password twice to reset it.
Still need to integrate it with the frontend guys once they create the page.
There is some reformatting and variable setting left but we will clean up the whole project soon.

DraganMitrasinovicHtec committed:
Successfully sending password reset emails, minor refactor

DraganMitrasinovicHtec committed:
Rebased, fixed conflicts

DraganMitrasinovicHtec committed:
Put endpoint for password reset after the email was sent

DraganMitrasinovicHtec committed:
Put endpoint for password reset after the email was sent

DraganMitrasinovicHtec committed:
Added password validation when resetting the password

DraganMitrasinovicHtec committed:
Added password validation when resetting the password

ivanfhtec committed:
SJQ2T-55: Always checkout bash scripts with LF

* added a .gitattributes

markobarackovhtec commented on backend/src/main/java/com/htecgroup/skynest/service/impl/UserServiceImpl.java:
```diff
@@ -64,14 +64,41 @@ public void sendVerificationEmail(String emailAddress) {
+
+  @Override
+  public String resetPassword(String token, String password) {
+    boolean validateToken = jwtUtils.validateJwtToken(token);
```
Maybe method validateJwtToken can throw exception if token is not valid and to be void, instead boolean ?

DraganMitrasinovicHtec replied:
That makes sense, will make the method look cleaner as it won't have the if statement and exception throwing. Will change

markobarackovhtec commented on backend/src/main/java/com/htecgroup/skynest/util/JwtUtils.java:
```diff
@@ -132,4 +138,77 @@ public String buildEmail(String email, String link) {
   }
+
+  public String buildPasswordResetEmail(String email, String link) {
+    return "<div style=\"font-family:Helvetica,Arial,sans-serif;font-size:16px;margin:0;color:#0b0c0c\">\n"
```
Is there a way to move this huge String in some file, and to read it from file ?

DraganMitrasinovicHtec replied:
There should be, we plan to do the same with the verification email.

markobarackovhtec replied:
Then create a ticket for that, so that you dont forget it :)

DraganMitrasinovicHtec replied:
I made sure to add it as part of the refactor task checklist

ivanfhtec committed:
SJQ2T-65: hotfix CORS

* allow FE's origin for every endpoint.

* allow unauthenticated access to swagger

DraganMitrasinovicHtec committed:
Changed verifyToken to void
Deleted now irrelevant tests
Changed url path naming to hyphens

DraganMitrasinovicHtec committed:
Changed verifyToken to void
Deleted now irrelevant tests
Changed url path naming to hyphens

DraganMitrasinovicHtec commented:
Changed validateToken to void
Deleted now irrelevant tests
Changed url path naming to hyphens
Writing a comment because github doesn't show the full title

DraganMitrasinovicHtec committed:
Merge remote-tracking branch 'origin/SJQ2T-21_Forgot_Password_Functionality' into SJQ2T-21_Forgot_Password_Functionality

# Conflicts:
#	backend/src/main/java/com/htecgroup/skynest/security/WebSecurity.java

markobarackovhtec commented on backend/src/main/java/com/htecgroup/skynest/util/JwtUtils.java:
```diff
@@ -0,0 +1,212 @@
+      JWTVerifier verifier = verification.acceptExpiresAt(expiresAt.getTime()).build();
+      verifier.verify(token);
+    } catch (JWTVerificationException e) {
+      log.error("Invalid JWT token: {}", e.getMessage());
```
Throw exception here, i believe that you need to return error to end user if token is not valid ?

DraganMitrasinovicHtec replied:
Makes sence, so the front-end can show them a pop-up or something.

markobarackovhtec commented on backend/src/main/java/com/htecgroup/skynest/util/JwtUtils.java:
```diff
@@ -0,0 +1,212 @@
+    } catch (JWTVerificationException e) {
+      log.error("Invalid JWT token: {}", e.getMessage());
+    } catch (IllegalArgumentException e) {
+      log.error("JWT claims string is empty: {}", e.getMessage());
```
same

markobarackovhtec commented on backend/src/main/java/com/htecgroup/skynest/service/impl/UserServiceImpl.java:
```diff
@@ -86,14 +111,12 @@ public UserDto findUserByEmail(String email) {
-    boolean validateToken = jwtEmailVerificationUtils.validateJwtToken(token);
-    if (validateToken) {
-      String email = jwtEmailVerificationUtils.getEmailFromJwtEmailVerificationToken(token);
+    jwtUtils.validateJwtToken(token);
```
Not formatted correctly

DraganMitrasinovicHtec replied:
My bad, thought I turned on autoformat on commit.

markobarackovhtec commented on backend/src/main/java/com/htecgroup/skynest/service/impl/UserServiceImpl.java:
```diff
@@ -64,14 +64,39 @@ public void sendVerificationEmail(String emailAddress) {
+
+  @Override
+  public String resetPassword(String token, String password) {
+    jwtUtils.validateJwtToken(token);
```
Not formatted correctly

DraganMitrasinovicHtec replied:
Will clean it up

DraganMitrasinovicHtec committed:
Formatted code, throwing exceptions on invalid/missing tokens
